packages/image_picker/image_picker/example/lib/main.dart

```diff
@@ -69,29 +69,32 @@ class _MyHomePageState extends State<MyHomePage> {
     if (_controller != null) {
       await _controller.setVolume(0.0);
     }
-    if (isVideo) {
-      final PickedFile file = await _picker.getVideo(
-          source: source, maxDuration: const Duration(seconds: 10));
-      await _playVideo(file);
-    } else {
-      await _displayPickImageDialog(context,
-          (double maxWidth, double maxHeight, int quality) async {
-        try {
-          final pickedFile = await _picker.getImage(
-            source: source,
-            maxWidth: maxWidth,
-            maxHeight: maxHeight,
-            imageQuality: quality,
-          );
-          setState(() {
-            _imageFile = pickedFile;
-          });
-        } catch (e) {
-          setState(() {
-            _pickImageError = e;
-          });
-        }
-      });
+    switch (mediaSource) {
```

Comment:
I am not a fan of this global controlling the behavior of this method.

Do you think it'd be possible to split `_onImageButtonPressed` in two methods: `_onImageButtonPressed` and `_onVideoButtonPressed`? (Also I've seen other parts of the code using this global-but-not-quite-part-of-the-state variable)

(Anyway, that might be a refactor for another time)